src/Passes/Source/ValidationPass/ValidationPass.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "ValidationPass/ValidationPass.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+
+namespace microsoft
+{
+namespace quantum
+{
+    llvm::PreservedAnalyses ValidationPass::run(llvm::Module& module, llvm::ModuleAnalysisManager& /*mam*/)
+    {
+
+        for (auto& function : module)
+        {
+            for (auto& block : function)
+            {
+                for (auto& instr : block)
+                {
+                    auto opname = instr.getOpcodeName();
+                    if (opcodes_.find(opname) != opcodes_.end())
+                    {
+                        ++opcodes_[opname];
```

Comment:
Do we really need to count them? Looks like only presence is tested later.
If we don't need to count - consider unordered_set.
If we need to count consider wrapping this pattern into function (increase if present, set to 1 if not).

Reply:
It is a good suggestion. The ValidationPass is still work in progress and at this stage, it is unclear whether there are any merits to counting opcodes vs. flagging them.  Hence, I will leave it as it is for now until we understand the final architecture of the validation process better.